File: VB/DateNavigatorQueryActiveViewType_Example/Form1.vb
Imports Microsoft.VisualBasic
#Region "#usings"
Imports DevExpress.XtraScheduler
Imports System.Windows.Forms
#End Region ' #usings

Namespace DateNavigatorQueryActiveViewType_Example
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			schedulerControl1.OptionsView.FirstDayOfWeek = DevExpress.XtraScheduler.FirstDayOfWeek.Monday
			AddHandler schedulerControl1.DateNavigatorQueryActiveViewType, AddressOf schedulerControl1_DateNavigatorQueryActiveViewType
		End Sub
		#Region "#DateNavigatorQueryActiveViewType"
		Private Sub schedulerControl1_DateNavigatorQueryActiveViewType(ByVal sender As Object, ByVal e As DateNavigatorQueryActiveViewTypeEventArgs)
			If (e.SelectedDays.Count = 7) AndAlso (e.SelectedDays.Start.DayOfWeek = schedulerControl1.FirstDayOfWeek) Then
				e.NewViewType = SchedulerViewType.Week
			End If
		End Sub
		#End Region ' #DateNavigatorQueryActiveViewType
	End Class
End Namespace

File: CS/DXApplication201/Form1.cs
#region #usings
using DevExpress.XtraScheduler;
using System.Windows.Forms;
#endregion #usings

namespace DXApplication201
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            schedulerControl1.OptionsView.FirstDayOfWeek = DevExpress.XtraScheduler.FirstDayOfWeek.Monday;
            schedulerControl1.DateNavigatorQueryActiveViewType += schedulerControl1_DateNavigatorQueryActiveViewType;
        }
        #region #DateNavigatorQueryActiveViewType
        void schedulerControl1_DateNavigatorQueryActiveViewType(object sender, DateNavigatorQueryActiveViewTypeEventArgs e)
        {
            if ((e.SelectedDays.Count == 7) && (e.SelectedDays.Start.DayOfWeek == schedulerControl1.FirstDayOfWeek))
            {
                e.NewViewType = SchedulerViewType.Week;
            }
        }
        #endregion #DateNavigatorQueryActiveViewType
    }
}
